plugin/plugin.go

```diff
@@ -0,0 +1,66 @@
+package plugin
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"runtime"
+)
+
+const (
+	// NamePrefix is the prefix required on all plugin binary names.
+	NamePrefix = "notation-"
+
+	// MetadataSubcommandName is the name of the plugin subcommand
+	// which must be supported by every plugin and returns the
+	// plugin metadata.
+	MetadataSubcommandName = "get-plugin-metadata"
```

Comment:
Isn't this `discover`?

Reply:
We changed it to `get-plugin-metadata` in https://github.com/notaryproject/notaryproject/pull/150